src/main/java/org/folio/inventory/instanceingress/handler/CreateInstanceIngressEventHandler.java

```diff
@@ -0,0 +1,240 @@
+    instanceAsJson.put(SOURCE_KEY, sourceType.value());
+    if (isNull(instanceAsJson.getString(INSTANCE_TYPE_ID_KEY))) {
+      //q is that ok?
+      LOGGER.warn("No instance type provided in mapped instance with id '{}', setting 'unspecified'", instance.getId());
+      instanceAsJson.put(INSTANCE_TYPE_ID_KEY, "30fffe0e-e985-4144-b2e2-1e8179bdb41f");
+    }
+    return instanceAsJson;
+  }
+
+  private Future<Void> saveInstance(Instance instance, InstanceIngressEvent event) {
+    LOGGER.info("Saving Instance from InstanceIngressEvent with id '{}':", event.getId());
+    var targetRecord = (Record) event.getEventPayload().getAdditionalProperties().get(MARC_BIBLIOGRAPHIC.value());
+    var sourceContent = targetRecord.getParsedRecord().getContent().toString();
+    super.addInstance(instance, instanceCollection)
+      .compose(createdInstance -> getPrecedingSucceedingTitlesHelper().createPrecedingSucceedingTitles(instance, context).map(createdInstance))
+      .compose(createdInstance -> executeFieldsManipulation(createdInstance, targetRecord, event))
+      .compose(createdInstance -> {
+        var targetContent = targetRecord.getParsedRecord().getContent().toString();
+        var content = reorderMarcRecordFields(sourceContent, targetContent);
+        targetRecord.setParsedRecord(targetRecord.getParsedRecord().withContent(content));
+        return saveRecordInSrsAndHandleResponse(event, targetRecord, createdInstance);
+      });
+    return Future.succeededFuture();
+  }
+
+  private Future<Instance> executeFieldsManipulation(Instance instance, Record srcRecord, InstanceIngressEvent event) {
+    LOGGER.info("executeFieldsManipulation for an Instance with id '{}':", instance.getId());
+    AdditionalFieldsUtil.fill001FieldInMarcRecord(srcRecord, instance.getHrid());
+    if (StringUtils.isBlank(srcRecord.getMatchedId())) {
+      srcRecord.setMatchedId(srcRecord.getId());
+    }
+    super.setExternalIds(srcRecord, instance);
+    boolean bibframeIdSetTo999 = AdditionalFieldsUtil.addFieldToMarcRecord(srcRecord, TAG_999, SUBFIELD_B,
+      event.getEventPayload().getSourceRecordIdentifier());
+    boolean instanceIdSetTo999 = AdditionalFieldsUtil.addFieldToMarcRecord(srcRecord, TAG_999, SUBFIELD_I, instance.getId());
+    return bibframeIdSetTo999 && instanceIdSetTo999
+      ? Future.succeededFuture(instance)
+      : Future.failedFuture(format("Failed to add instance id '%s' to record with id '%s'", instance.getId(), srcRecord.getId()));
+  }
+
+  protected Future<Instance> saveRecordInSrsAndHandleResponse(InstanceIngressEvent event, Record srcRecord, Instance instance) {
+    LOGGER.info("Saving record in SRS and handling a response for an Instance with id '{}':", instance.getId());
+    Promise<Instance> promise = Promise.promise();
+    getSourceStorageRecordsClient(context.getOkapiLocation(), context.getToken(), context.getTenantId())
+      .postSourceStorageRecords(srcRecord)
+      .onComplete(ar -> {
```

Comment:
suggest to move it to separate method

Reply:
It's a copy-paste of AbstractInstanceEventHandler#saveRecordInSrsAndHandleResponse, I suggest to leave it as is, to keep it similar and comparable with the original